ecs-cli/modules/cli/cluster/cluster_app.go

```diff
@@ -400,6 +418,36 @@ func retrieveInstanceType(cfnParams *cloudformation.CfnStackParams) (string, err
 	return aws.StringValue(param.ParameterValue), nil
 }
 
+func populateInstanceTypeParameter(cfnParams *cloudformation.CfnStackParams) (string, error) {
+	param, err := cfnParams.GetParameter(ParameterKeyInstanceType)
+	if err == cloudformation.ParameterNotFoundError {
+		logrus.Infof("Defaulting instance type to %s", cloudformation.DefaultECSInstanceType)
+
+		cfnParams.Add(ParameterKeyInstanceType, cloudformation.DefaultECSInstanceType)
+
+		return cloudformation.DefaultECSInstanceType, nil
+	} else if err != nil {
+		return "", err
+	}
+
+	return aws.StringValue(param.ParameterValue), err
+}
+
+func validateInstanceType(instanceType string, supportedInstanceTypes []string) error {
+	found := false
+	for _, it := range supportedInstanceTypes {
+		if it == instanceType {
+			found = true
+			break
+		}
+	}
+	if !found {
+		return fmt.Errorf(invalidInstanceTypeFmt, instanceType, supportedInstanceTypes)
```

Comment:
I wonder if we need to output the whole supported instanceType because that will be huge. I would make this func return boolean and handle the "not found" error out of the function.

Reply:
It's unwieldy I agree, but it's helpful to the user, otherwise they'll just be asking themselves "well what are the supported instance types". I think it should stay.

Reply:
:+1: